cpp/tests/c_api/random_walks_test.c

```diff
@@ -0,0 +1,434 @@
+  runtime_assert(ret_code == CUGRAPH_SUCCESS, "dst device_buffer update failed.");
+
+  ret_code = cugraph_update_device_buffer(p_handle, weight_tid, &dbuf_wgt, (byte_t*)h_wgt);
+  runtime_assert(ret_code == CUGRAPH_SUCCESS, "weight device_buffer update failed.");
+
+  ret_code = cugraph_make_device_buffer(p_handle, vertex_tid, num_paths, &dbuf_start);
+  runtime_assert(ret_code == CUGRAPH_SUCCESS, "start device_buffer creation failed.");
+
+  ret_code = cugraph_update_device_buffer(p_handle, vertex_tid, &dbuf_start, (byte_t*)h_start);
+  runtime_assert(ret_code == CUGRAPH_SUCCESS, "start device_buffer update failed.");
+
+  p_sampling_strategy = cugraph_create_sampling_strategy(0, 0.0, 0.0);
+  runtime_assert(p_sampling_strategy != NULL, "start device_buffer update failed.");
+
+  p_graph_envelope = cugraph_make_sg_graph(p_handle,
+                                           vertex_tid,
+                                           edge_tid,
+                                           weight_tid,
+                                           FALSE,
+                                           &dbuf_src,
+                                           &dbuf_dst,
+                                           &dbuf_wgt,
+                                           num_vertices,
+                                           num_edges,
+                                           FALSE,
+                                           FALSE,
+                                           FALSE);
+  runtime_assert(p_graph_envelope != NULL, "graph envelope creation failed.");
+
+  ret_code = cugraph_random_walks(p_handle,
+                                  p_graph_envelope,
+                                  &dbuf_start,
+                                  num_paths,
+                                  max_depth,
+                                  flag_use_padding,
+                                  p_sampling_strategy,
+                                  &rw_ret);
+  runtime_assert(ret_code == CUGRAPH_SUCCESS, "cugraph_random_walks() failed.");
+
+  cugraph_free_rw_result(&rw_ret);
+
+  cugraph_free_graph(p_graph_envelope);
+
+  cugraph_free_sampling_strategy(p_sampling_strategy);
+
+  cugraph_free_device_buffer(&dbuf_start);
+
+  cugraph_free_device_buffer(&dbuf_wgt);
+
+  cugraph_free_device_buffer(&dbuf_dst);
+
+  cugraph_free_device_buffer(&dbuf_src);
+
+  cugraph_free_handle(p_handle);
+
+  return 0;
+}
+
+/* array of compressed paths,
+   using 1-based indexing for vertices,
+   to avoid confusion between, for example,
+   `012` and `12`, which result in same number*/
+#define NUM_MAX_PATHS 30
+static int32_t c_ps_array[NUM_MAX_PATHS] = {
+  1,  2,  3,  4,   5,    6,    12,    124,   125, 1246, 1256, 24,   25,  246, 256,
+  31, 32, 34, 312, 3124, 3125, 31246, 31256, 324, 325,  3246, 3256, 346, 46,  56};
+
+/* linear search of `value` inside `p_cmprsd_path[max_num_paths]`*/
+bool_t is_one_of(int32_t value, int32_t* p_cmprsd_path, int max_num_paths)
```

Comment:
I think some of these abbreviations are harder to type and read than the non-abbreviated words.  Can this just say "compressed" instead of "cmprsd" (assuming that's what it's abbreviating)?

Reply:
Again, question of style/taste. You're free to change them in the next release.